cogs/displayinfo.py

```diff
@@ -0,0 +1,131 @@
+mongoClient = MongoClient(uri, server_api=ServerApi('1'))
+
+
+try:
+    # Send a ping to confirm a successful connection to MongoDB
+    mongoClient.admin.command('ping')
+    print("Pinged your deployment. You successfully connected to MongoDB!")
+except Exception as e:
+    print(e)
+
+# Select the database
+db = mongoClient.Rasenbot
+
+
+
+class DisplayInfo(commands.Cog):
+
+    def __init__(self,client):
+        self.client = client
+        self.timers = {}
+        # self.tracked_games = ['Rocket League', 'League of Legends', "Spotify"]
+        self.current_collection = ""
+        self.count = 0
+        self.current_date = str(date.today())
+
+    
+    # @commands.Cog.listener()
+    # async def on_ready(self):
+    #     for guild in self.client.guilds:
+    #         guild_name = guild.name
+
+    #         # Check if the MongoDB collection for the guild already exists
+    #         if guild_name not in db.list_collection_names():
+    #             # If the collection doesn't exist, create a new one for the guild
+    #             newCollection = db[guild_name]
+    #             self.current_collection = newCollection
+    #             newCollection.insert_one({})  # Insert a dummy document to initialize the collection
+    #             self.timers = self.current_collection.find_one()
+    #         else:
+    #             # If the collection exists, get the existing collection for the guild
+    #             self.current_collection = db[guild_name]
+    #             # Load existing timers (if any) from the MongoDB collection
+    #             self.timers = self.current_collection.find_one()
+    
+
+
+    # The 'before' parameter represents the member's previous state, and 'after' represents the current state.
+    @commands.Cog.listener()
+    async def on_member_update(self, before:discord.Member, after:discord.Member):
+
+        # Initialize the MongoDB collection for the guild if it doesn't exist
+        if self.count == 0:
+            if f"{before.guild.name}" not in db.list_collection_names():
+                newCollection = db[f"{before.guild.name}"]  # If the collection doesn't exist, create a new one for the guild/server
+                self.current_collection = newCollection               
+                newCollection.insert_one({}) # Create a new blank document in the new Collection
+                self.timers = self.current_collection.find_one() # Load blank document from the MongoDB collection
+            else:
+                self.current_collection = db[f"{before.guild.name}"] # If the collection exists, get the existing collection for the guild
+                self.timers = self.current_collection.find_one() # Load existing document from the MongoDB collection
+            self.count += 1 # Increment the counter to prevent further initialization for this guild
+        
+        # Track activity updates for members
+        if not before.activity and after.activity:
```

Comment:
Comment differentiating transitions of activity --> no activity vs no activity --> activity will be helpful to separate the ifs.